### PR TITLE
Fix OOB

### PR DESCRIPTION
**Pull request by shargon**
Close https://github.com/neo-project/neo-vm/issues/196

### CODE REVIEW

File: src/neo-vm/Unsafe.cs [hunk 1]
```diff
@@ -36,6 +36,10 @@ public static bool MemoryEquals(byte[] x, byte[] y)
         public static void MemoryCopy(byte[] src, int srcOffset, byte[] dst, int dstOffset, int count)
         {
             if (count == 0) return;
+            if (srcOffset < 0 || count < 0 || dstOffset < 0) throw new ArgumentException();
+            if (checked(srcOffset + count) > src.Length) throw new ArgumentException();
+            if (checked(dstOffset + count) > dst.Length) throw new ArgumentException();
```
Comment (erikzhang) [inline src/neo-vm/Unsafe.cs]: The `unsafe` methods shouldn't check these parameters. The responsibility for checking should be at the caller.

File: src/neo-vm/ExecutionEngine.cs [hunk 2]
```diff
@@ -450,6 +450,7 @@ private bool ExecuteInstruction()
                         {
                             int count = (int)context.EvaluationStack.Pop().GetBigInteger();
                             if (count < 0) return false;
+                            if (count > MaxItemSize) count = (int)MaxItemSize;
```
Comment (shargon) [inline src/neo-vm/ExecutionEngine.cs]: return false?

Comment (erikzhang) [reply src/neo-vm/ExecutionEngine.cs]: https://github.com/neo-project/neo-vm/blob/cff92bb3ce36a377fa089b13d38d2f0d8745e077/src/neo-vm/ExecutionEngine.cs#L457

**Comment by shargon** [comment]
@erikzhang It can be merged?

**Pull request merged**